### PR TITLE
[Storage] Fix#21414: `az storage blob sync`: Fix the flag `--delete-destination` default to false

### LINKED ISSUE #21414: az storage blob sync's deletion behavior can lead to unexpected data loss
**Is your feature request related to a problem? Please describe.**

Currently, the `az storage blob sync` command will delete blobs at the destination that are not present in the source. This is hardcoded into the command, with no documentation of its behavior or an option to change it. Additionally, this is inconsistent with the default behavior of other prominent tools for Azure as well as other cloud providers' storage services, including `azcopy`, which [include a flag to enable deleting but do not do so by default](https://docs.microsoft.com/en-us/azure/storage/common/storage-ref-azcopy-sync#options). 

In fact, this command is implemented in `azure-cli` by wrapping `azcopy sync` but then overriding `azcopy`'s default. Relevant source code: 

https://github.com/Azure/azure-cli/blob/92c50bc2f847907b0628dfc3bd885bac05b96c36/src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py#L84-L87 

This is a risky design that can lead to unexpected deletion by users and result in data loss. (This happened to me.)

**Describe the solution you'd like**

1. Add a `--delete-destination ` to match the functionality of `azcopy sync`.
2. Have that flag default to false.

### PR DESCRIPTION
**Pull request by jonie001**
…ation default to false

**Description**

Fix #21414 

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

**Comment by evelyn-ys** [comment]
Changing the default value is breaking change. How about add a parameter `--delete-destination` which defaults to `True`? For customers complaining about this behavior, they can customize to `Flase` by themselves.

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -1256,6 +1259,7 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('source', options_list=['--source', '-s'],
                    help='The source file path to sync from.')
         c.ignore('destination')
+        c.argument('delete_destination', delete_destination_type)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py:1262]: Try to use `get_three_state_flag` arg type

File: src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py [hunk 2]
```diff
@@ -81,10 +81,14 @@ def storage_fs_directory_copy(cmd, source, destination, recursive=None, **kwargs
     azcopy.copy(source, destination, flags=flags)
 
 
-def storage_blob_sync(cmd, client, source, destination, exclude_pattern=None, include_pattern=None,
-                      exclude_path=None):
+def storage_blob_sync(cmd, client, source, destination, delete_destination=None, exclude_pattern=None,
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py:84]:
```suggestion
def storage_blob_sync(cmd, client, source, destination, delete_destination=True, exclude_pattern=None,
```
In this way, `default=True` will automatically show in help message

File: src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py [hunk 3]
```diff
@@ -81,10 +81,14 @@ def storage_fs_directory_copy(cmd, source, destination, recursive=None, **kwargs
...
-def storage_blob_sync(cmd, client, source, destination, exclude_pattern=None, include_pattern=None,
-                      exclude_path=None):
+def storage_blob_sync(cmd, client, source, destination, delete_destination='true', exclude_pattern=None,
+                      include_pattern=None, exclude_path=None):
     azcopy = _azcopy_blob_client(cmd, client)
-    flags = ['--delete-destination=true']
+    flags = []
+    if delete_destination == 'true':
+        flags.append('--delete-destination=true')
+    else:
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py:90]:
```suggestion
    if delete_destination is not None:
```

**Pull request merged**